src/language-js/needs-parens.js

```diff
@@ -288,6 +288,24 @@ function needsParens(path, options) {
         return true;
       }
       break;
+
+    case "VariableDeclarator":
+      // Legacy syntax
+      // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Invalid_for-in_initializer
+      // `for (var a = 1 in b);`
+      if (
+        key === "init" &&
+        path.match(
+          undefined,
+          undefined,
+          (node, key) =>
+            key === "declarations" && node.type === "VariableDeclaration",
+          (node, key) => key === "left" && node.type === "ForInStatement",
+        )
+      ) {
+        return true;
+      }
+      break;
```

Comment:
@sosukesuzuki I just realized this change may affect MDN docs, I don't think people use this feature, but MDN does, should we revert?

Reply:
@Josh-Cena I decided to add `()` to all nodes in `for...in` initializer, since there are broken cases, I thought no body actually use this feature, but MDN docs does, is it fine for you to ignore?

Reply:
Yeah, not a problem for us. We can either ignore or just live with it. Thanks for notifying me.